srml/support/src/lib.rs

```diff
@@ -326,6 +326,35 @@ mod tests {
 		});
 	}
 
+	fn double_map_swap_works() {
+		with_externalities(&mut new_test_ext(), || {
+			type DoubleMap = DataDM;
+			DataDM::insert(0, 1, 0);
+			DataDM::insert(0, 2, 1);
+			DataDM::insert(0, 3, 2);
+			DataDM::insert(0, 4, 3);
+
+			let collect = || DataDM::enumerate().collect::<Vec<_>>();
```

Comment:
We don't have `enumerate()` for double maps.

Reply:
What does `DataDM` have available? I see that it's generated by a `decl_storage!` macro, a little hard to know what it actually is.

Reply:
decl_storage doc is [here](https://substrate.dev/rustdocs/master/srml_support/macro.decl_storage.html)

so it implements StorageDoubleMap

Reply:
Hm, I see - so we can't iterate over all `(key1, key2, value)` triplets.